File: 58. Length of Last Word/main.py
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        count = 0
        next = " "

        for c in s[::-1]:
            if next != " " and c == " ":
                break

            if c != " ":
                count += 1

            next = c

        return count


class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        return len(s.split()[-1])

File: 58. Length of Last Word/main.rs
struct Solution;

impl Solution {
    pub fn length_of_last_word(s: String) -> i32 {
        s.split_whitespace().last().unwrap().len() as i32
    }
}

impl Solution {
    pub fn length_of_last_word(s: String) -> i32 {
        let mut count = 0;
        let mut next = ' ';

        for c in s.chars().rev() {
            if next != ' ' && c == ' ' {
                break;
            }

            if c != ' ' {
                count += 1;
            }

            next = c;
        }

        count
    }
}
